### PR TITLE
Git: patch to make git relocatable with buildcache.

### PR DESCRIPTION
**Pull request by gartung**
Apply patch to git to make it search for config files from its current install directory instead of the hard coded at install directory.

### CODE REVIEW

**Comment by gartung** [comment]
@chissg

**Comment by adamjstewart** [comment]
Where is the source of this patch? Did you write it yourself? Has it been submitted upstream?

**Comment by gartung** [comment]
It is a patch used at FNAL to build relocatable git
https://cdcvs.fnal.gov/redmine/projects/build-framework/repository/git-ssi-build/revisions/master/raw/patch/git.patch

**Comment by adamjstewart** [comment]
Can you add that URL next to the patch so we know where it came from? I wonder if it has been submitted upstream though. Developers hate it when people patch their software to fix bugs but never tell them about it...

**Comment by gartung** [comment]
OK I added the provenance info to the patch file.

**Comment by gartung** [comment]
Ah I see our script does
make NO_INSTALL_HARDLINKS=1 ETC_GITATTRIBUTES=config ETC_GITCONFIG=config all install 
Any idea how to do this without overriding the install method from autotools?

**Comment by gartung** [comment]
https://cdcvs.fnal.gov/redmine/projects/build-framework/repository/git-ssi-build/revisions/master/entry/build_git.sh#L107

**Comment by adamjstewart** [comment]
@gartung You can do the following:
```python
build_targets = ['NO_INSTALL_HARDLINKS=1', 'ETC_GITATTRIBUTES=config', 'ETC_GITCONFIG=config']
install_targets = ['NO_INSTALL_HARDLINKS=1', 'ETC_GITATTRIBUTES=config', 'ETC_GITCONFIG=config', 'install']
```
or you can override the `build` and `install` methods if you prefer.

File: var/spack/repos/builtin/packages/git/package.py [hunk 1]
```diff
@@ -236,6 +238,16 @@ def filter_rt(self):
...
 
+    @run_after('configure')
+    def make_opts(self):
+        self.build_targets=['NO_INSTALL_HARDLINKS=1',
+                            'ETC_GITATTRIBUTES=config',
+                            'ETC_GITCONFIG=config']
+        self.install_targets=['NO_INSTALL_HARDLINKS=1',
+                              'ETC_GITATTRIBUTES=config',
+                              'ETC_GITCONFIG=config',
+                              'install']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/git/package.py]: These don't need to be inside of a method, they can just be global variables defined inside the class.

**Pull request merged**